src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -97,75 +91,54 @@ public function load(mixed $resource, ?string $type = null): mixed
-            }
-
-            foreach ($result ?? [] as $key => $config) {
-                if (!str_starts_with($key, 'when@')) {
-                    $config = [$key => $config];
-                } elseif (!$this->env || 'when@'.$this->env !== $key) {
-                    continue;
-                } elseif ($config instanceof ServicesConfig || $config instanceof ParametersConfig || $config instanceof ImportsConfig || $config instanceof ConfigBuilderInterface) {
-                    $config = [$config];
-                } elseif (!is_iterable($config)) {
-                    throw new InvalidArgumentException(\sprintf('The "%s" key should contain an array in "%s".', $key, $path));
-                }
-
-                foreach ($config as $key => $config) {
-                    $expectedKey = match (true) {
-                        $config instanceof ServicesConfig => 'services',
-                        $config instanceof ParametersConfig => 'parameters',
-                        $config instanceof ImportsConfig => 'imports',
-                        default => null,
-                    };
-                    if (null !== $expectedKey || \in_array($key, ['services', 'parameters', 'imports'], true)) {
-                        if (null === $expectedKey) {
-                            $config = [$key => $config];
-                        } elseif (\is_string($key) && $key !== $expectedKey) {
-                            throw new InvalidArgumentException(\sprintf('Invalid key "%s" returned for the "%s" config builder; none or "%s" expected in file "%s".', $key, get_debug_type($config), $expectedKey, $path));
-                        } else {
-                            $config = [$expectedKey => $config->config];
-                        }
-                        $yamlLoader = new YamlFileLoader($this->container, $this->locator, $this->env, $this->prepend);
-                        $yamlLoader->setResolver(new LoaderResolver([$this]));
-                        $loadContent = new \ReflectionMethod(YamlFileLoader::class, 'loadContent');
-                        ++$this->importing;
-                        try {
-                            $loadContent->invoke($yamlLoader, ContainerConfigurator::processValue($config), $path);
-                        } finally {
-                            --$this->importing;
+                $this->callConfigurator($result, new ContainerConfigurator($this->container, $this, $this->instanceof, $path, $resource, $this->env), $path);
+            } elseif (\is_array($result)) {
+                $yamlLoader = new YamlFileLoader($this->container, $this->locator, $this->env, $this->prepend);
+                $yamlLoader->setResolver($this->resolver ?? new LoaderResolver([$this]));
+                $loadContent = new \ReflectionMethod(YamlFileLoader::class, 'loadContent');
```

Comment:
I suggest extracting this into a dedicated (internal) class with a public method instead of accessing a private method of the YamlFileLoader through Reflection (and then also having to remember that error messages in code called from that method must not mention Yaml as this YamlFileLoader code could be used for non-Yaml config)